.github/workflows/rust.yml

```diff
@@ -12,6 +12,24 @@ defaults:
     working-directory: ./rust
 
 jobs:
+  static-analysis:
+    runs-on:
+      - ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Update toolchain
+        run: rustup show
+      - uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: ./rust
+          save-if: ${{ github.ref == 'refs/heads/main' }}
+      - run: cargo fmt -- --check
+      - run: |
+          cargo doc --all-features --no-deps --document-private-items ${{ matrix.packages }}
+        env:
+          RUSTDOCFLAGS: "-D warnings"
+      - run: |
+          cargo clippy --all-targets --all-features ${{ matrix.packages }} -- -D warnings
```

Comment:
The reason we had this in the `test` job was because we wanted clippy to lint on all platforms to ensure we lint also platform dependent code.

Reply:
Ah I see -- thanks for the catch. Would it make sense to just add a matrix for this job then?